src/ameba.ts

```diff
@@ -29,22 +30,29 @@ export class Ameba {
         this.config = getConfig();
     }
 
-    public execute(document: TextDocument, virtual: boolean = false): void {
-        if (!isCrystalDocument(document)) return;
-        if (isDocumentVirtual(document) && !virtual) return;
+    public execute(document: TextDocument | WorkspaceFolder, virtual: boolean = false): void {
+        const isWorkspace = !('languageId' in document);
```

Comment:
Why having this weird check if you know that the `document` is an instance of `WorkspaceFolder`?

Reply:
So that the typescript compiler is happy. This is the only way I could figure out to check if `document` implements TextDocument or WorkspaceFolder, instead of both.

Reply:
I'm taking about the `const isWorkspace = !('languageId' in document)`.
Why not `const isWorkspace = document instanceof WorkspaceFolder`?

Reply:
I know, that's what I was referring to. `WorkspaceFolder` is an interface, so `instanceof` doesn't work:

![image](https://github.com/user-attachments/assets/1778648f-4cfc-4064-be85-9373b1884d8d)

Reply:
In this case I believe it's better to use a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)